src/pages/index.js

```diff
@@ -4,42 +4,145 @@ import FormValidator from "../components/FormValidator.js";
 import UserInfo from "../components/UserInfo.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupDelete from "../components/PopupDelete.js";
 import { data } from "autoprefixer";
 import {
-  initialCards,
   profileAddButton,
   profileEditButton,
+  avatarEditButton,
+  profileSaveButton,
+  cardCreateSaveButton,
+  avatarSaveButton,
   config,
+  newName,
+  newJob,
   formData,
+  cardElement,
 } from "../Utility/Constant.js";
 
-function handleProfileFormSubmit(formData) {
-  userInfo.setUserInfo(formData);
-  profileEditPopup.close();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "266f9e9c-f310-4db4-b30c-94bce5a357d4",
+    "Content-Type": "application/json",
+  },
+});
+
+let section;
+let userInfo;
+
+api.userInfoAndCards();
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#cards-template",
+    handleImageClick,
+    handleCardDelete,
+    handleCardLike
+  );
+  return card.generateCard();
 }
+api.getInitialCards().then((data) => {
+  section = new Section({ items: data, renderer: createCard }, ".cards__list");
+  section.renderItems();
+});
+api.getUser().then((data) => {
+  userInfo = new UserInfo(
+    ".profile__title",
+    ".profile__description",
+    ".profile__image"
+  );
+  userInfo.setUserInfo({ title: data.name, description: data.about });
+  userInfo.changeAvatarImage(data.avatar);
+});
 
-function handleProfileFormCreate(inputValues) {
-  const cardElement = createCard(inputValues);
-  section.addItem(cardElement);
-  profileCardPopup.close();
+function handleProfileFormSubmit(formData) {
+  profileSaveButton.textContent = "Saving...";
+  let profileData = { name: newName.value, about: newJob.value };
+  api
+    .editProfile(profileData)
+    .then(() => {
+      userInfo.setUserInfo(formData);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileSaveButton.textContent = "Save";
+    });
+}
+function handleProfileFormCreate({ name, link }) {
+  cardCreateSaveButton.textContent = "Saving...";

```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```